goldmane/pkg/aggregator/aggregator.go

```diff
@@ -34,19 +35,32 @@ const (
 	channelDepth = 5000
 )
 
-// Sink is an interface that can receive aggregated flows.
-type Sink interface {
-	Receive(*AggregationBucket)
-}
-
 // flowRequest is an internal helper used to synchronously request matching flows from the aggregator.
 type flowRequest struct {
-	respCh chan []*proto.Flow
+	respCh chan *flowResponse
 	req    *proto.FlowRequest
 }
 
+type flowResponse struct {
+	flows []*proto.Flow
+	err   error
+}
+
 type LogAggregator struct {
-	buckets []AggregationBucket
+	// indices allow for quick handling of flow queries sorted by various methods.
+	indicies map[proto.SortBy]Index[string]
```

Comment:
This is just by string right now since it's the only kind we have. Probably want separate maps for each sortable type we want, eventually:

```go
  stringIndicies map[proto.SortBy]Index[string]
  intIndicies map[proto.SortBy]Index[int]
```

Having them as a map makes it simple to lookup which index satisfies the given request, and allows us to iterate all indicies in places we need to - like adding/removing DiachronicFlow objs.